source-academy/cadet-frontend
# Display private testcases for staff during submission grading

## Display private testcases for staff during submission grading
Summary: Code quality improvements and front-end support for the minor feature request https://github.com/source-academy/cadet/issues/445.

### Related PRs
Please review this PR in conjunction with the PR on cadet: https://github.com/source-academy/cadet/pull/633

### Changelog
- Remove unnecessary React `key` prop from header elements in the Autograder
- Simplify construction of the class name applied to Autograder testcase cards
- Add CSS to distinguish private testcases from public testcases
  - Cards for private testcases have a gray background
  - When run, private testcase cards fade back to gray after the initial color fade signifying the testcase status
- Remove unnecessary background color fade for Autograder result cards
- Update snapshot and DOM tests for the side-content Autograder to include private Autograder testcases

### Verification instructions
The PR can be tested both with and without the development backend.

1. Navigate to the Grading page, select any submission to view and proceed to any **programming** (not MCQ) question
2. Bring up the side-content Autograder (airplane icon), and locate the private testcases with the gray background (if any)
3. Run them individually (by clicking on the testcase card) or as part of the entire set of testcases (by clicking the editor's 'Run' button or pressing Shift-Enter with focus on the editor)
4. If testing with the development backend, you can additionally navigate to the assessment (e.g. Mission) corresponding to the viewed submission and verify that the private testcases are **not available** in the Autograder

### Screenshots

#### Gray background for private Autograder testcases
![image](https://user-images.githubusercontent.com/44989315/87641798-31491780-c77b-11ea-8bb9-9d8bd74b4202.png)

Last updated 16 Jul 2020, 3:45PM
